core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -959,6 +966,35 @@ public void testLoadMissingTable() {
         .hasMessageStartingWith("Table does not exist: ns.tbl");
   }
 
+  @Test
+  public void testLoadTableWithMissingMetadatafile(@TempDir Path tempDir) throws IOException {
```

Comment:
`CatalogTests` is extended by multiple catalog suites. I think the change in class `ErrorHanlders` only applies to REST catalog. We may not need it to be triggered in non-rest catalog, like `TestHiveCatalog`, `TestBigQueryCatalog`

Reply:
Yes, the ErrorHandlers is specific to REST catalog, but the test would be applicable for all kind of catalogs. All other catalogs return NotFound exception